infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerMachine.java

```diff
@@ -148,6 +150,14 @@ void setServerStatus(String serverRef, ServerStatus status) {
         server.setStatus(status);
     }
 
+    public void putResource(String targetPath, InputStream sourceStream) throws InfrastructureException {
```

Comment:
I'd leave it as it is, because I prefer to have the same approach. And there are a lot of classes methods which could have package private visibility (like DockerMachineStarter, DockerMachineStarter etc) but there don't have.